tests/test_impls.py

```diff
@@ -322,3 +323,38 @@ def test_edit_pyproject(
 
         with open("pyproject.toml") as f:
             assert f.read() == original_contents
+
+
+@pytest.mark.parametrize(
+    ["setup_py_content", "expect_error"],
+    [
+        ("", False),
+        ("from setuptools import setup\n\nsetup()\n", False),
+        # 'setup_requires' in a comment on its own line
+        ("from setuptools import setup\n# setup_requires\n\nsetup()\n", False),
+        # 'setup_requires' actually passed into setup(), on the same line
+        ("from setuptools import setup\nsetup(setup_requires=[])\n", True),
+        # 'setup_requires' actually passed into setup(), on its own line
+        (
+            "from setuptools import setup\nsetup(\n    setup_requires=['rmm']\n)\n# setup_requires\n",  # noqa: E501
+            True,
+        ),
+        # 'setup_requires' actually passed into setup(), via a dictionary
+        (
+            "from setuptools import setup\nopts={'setup_requires': ['rmm']}\nsetup(**opts)\n",  # noqa: E501
+            True,
+        ),
```

Comment:
Not a big deal, but this parametrization is hard to read. You might consider doing something like creating a list of multiline strings or something and then using `zip(contents, errors)` instead.

Reply:
Yeah you're right. I just tried reformatting it by removing "should this raise an error?" from the parameterization completely and splitting it into 2 tests (one with cases that shouldn't result in an error, one with cases that should).

I think it's a lot easier to read and understand now:

<img width="671" alt="image" src="https://github.com/rapidsai/rapids-build-backend/assets/7608904/9ef5fd54-77f1-4916-9b9f-0f895a5bb4d9">